### PR TITLE
Sample sorting pr

### PR DESCRIPTION
**Pull request by gideonite**
In this PR I add the standard "Memo Sorting" algorithm which is a special case of a radix sort and is implemented as such. Compare the changes here to the MemoSort.js file in the old project. Much less code! Check out the commit messages for more discussion.

### CODE REVIEW

File: src/js/sorting.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+function rows_to_indexers(rows) {
+  return _.range(rows.length)
+  .reverse()     // least significant first
+  .map(function(ith_row) {
+    return function(index) { return rows[ith_row][index]; };
+  });
+}
+
+exports.genomic_metric = function genomic_metric(x) {
+  var cna_order = {AMPLIFIED:400, HOMODELETED:300, GAINED:200, HEMIZYGOUSLYDELETED:100, DIPLOID: 0, undefined: 0};
```
Comment (adamabeshouse) [inline src/js/sorting.js]: Might be cleaner to keep all of these single digit and then multiply it in the return value. So AMPLIFIED:4, HOMODELETED:3, ... and then return -1* (1000*cna_order[x.cna] + 100*regulated_order, ...)

File: src/js/sorting.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+  // need -1 to flip the order.
+  return -1 * (cna_order[x.cna]
+               + regulated_order[x.mrna]
+               + regulated_order[x.rppa]
+               + mutation_order_f(x.mutation));
+};
+
+// indexers is least significant first.
+exports.radix = function radix(datums, compare, indexers) {
```
Comment (adamabeshouse) [inline src/js/sorting.js]: exports.radixSort?

**Comment by adamabeshouse** [comment]
Besides small comments LGTM

File: src/js/sorting.js (hunk 1 above)
Comment (gideonite) [inline src/js/sorting.js]: Yes, good idea.

File: src/js/sorting.js (hunk 2 above)
Comment (gideonite) [inline src/js/sorting.js]: Sure, doesn't really make a difference to me.

**Pull request merged**